jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -353,7 +353,11 @@ protected HiveConnection(String uri, Properties info,
     }
     if (isEmbeddedMode) {
       client = EmbeddedCLIServicePortal.get(connParams.getHiveConfs());
+      String sessionId = connParams.getHiveConfs().get(HiveConf.ConfVars.HIVESESSIONID.varname);
```

Comment:
this part seems to be hacky, original code made a client from the confs then cleared the conf, now we're forcing the sessionId to be present in the hive confs after clearing, why is it? 
this way we'll have an almost empty connParams.getHiveConfs(), having only session id